rust/tool/src/esp.rs

```diff
@@ -70,6 +56,38 @@ impl EspPaths {
     }
 }
 
+/// Contains the generation-specific paths to the ESP.
```

Comment:
nit: I honestly had a bit of trouble making sense of this. Maybe:

```
/// Paths to the boot files of a specific generation.
///
/// All of these paths point into the ESP.
```

Reply:
I still have a hard time finding a good comment/explanation. Your suggestions are definetely an improvement though.